TradeItIosTicketSDK2/TradeItAlertManager.swift

```diff
@@ -63,14 +69,23 @@ import UIKit
                                      onViewController viewController: UIViewController,
                                      onAnswerSecurityQuestion: (withAnswer: String) -> Void,
                                      onCancelSecurityQuestion: () -> Void) {
-        let alertController = TradeItAlertProvider.provideSecurityQuestionAlertWith(
-            alertTitle: "Security Question",
-            alertMessage: securityQuestion.securityQuestion ?? "No security question provided.",
-            multipleOptions: securityQuestion.securityQuestionOptions ?? [],
-            alertActionTitle: "Submit",
-            onAnswerSecurityQuestion: onAnswerSecurityQuestion,
-            onCancelSecurityQuestion: onCancelSecurityQuestion)
-        viewController.presentViewController(alertController, animated: true, completion: nil)
+        add(alertClosure: {
+            let alert = TradeItAlertProvider.provideSecurityQuestionAlertWith(
+                alertTitle: "Security Question",
+                alertMessage: securityQuestion.securityQuestion ?? "No security question provided.",
+                multipleOptions: securityQuestion.securityQuestionOptions ?? [],
+                alertActionTitle: "Submit",
+                onAnswerSecurityQuestion: { answer in
+                    onAnswerSecurityQuestion(withAnswer: answer)
+                    self.activeAlertFinished()
```

Comment:
This is pretty funky. I'm not totally happy with having to remember to call `activeAlertFinished`. Perhaps the caching is better in the provider?